### DISCUSSION TITLE
Polygonal (6 or more points) objects?

### DISCUSSION THREAD

**Discussion by davorvr** [rupertoverall/Rtrack#8] [Q&A] [answered]
Hello, due to a slightly off-center camera, the objects in my novel object recognition videos are at an angle so a rectangle doesn't really encompass them nicely. Is it possible to specify a polygonal object? Six points would already let me define the perimeter pretty nicely.

**Comment by rupertoverall** [comment, marked as answer]
This feature is not yet built in to Rtrack I'm afraid. Rectangular objects can be defined by the coordinates of their 4 corners - that has two side effects: 1) if the _arena_ is defined by corners, it will be affine-transformed into a square. That is useful for dealing with off-centre cameras. 2) You can also define non-square objects (but only with exactly 4 points at the moment).
 
We can add a multi-point object boundary to a future release. I have tagged this as a 'feature request' so we remember to do that.

**Comment by davorvr** [comment]
@rupertoverall I have submitted an implementation of this functionality in #9

**Comment by rupertoverall** [comment]
Just to briefly reopen this discussion, there are two new features in Rtrack version 2.0.4 (now in CRAN) that address your issue:
1. Support for square arena correction.
2. Support for irregular polygonal objects.

In the use case you described, an off-centre camera, the first feature should be all you need. If the open field or novel object recognition arena (which must be square by definition) is not square in the video, define the arena bounds as a 'square' using the 'four corners' method (x,y coordinates for all four corners). Rtrack will then transform this arena _and all of the objects in it_ to be a standardised square. The paths of any animals exploring this arena will also be transformed using the same model. This should completely correct for the skewed view coming from an off-centre camera (a common issue, especially when recording multiple arenas simultaneously with one camera).
Please note that this skew correction is currently only available for square arenas. It is also possible for circular arenas but the definition of the ellipse is too complex without visual feedback. I am working on a web-based tool to make complex arena definition more straightforward (this will be too fiddly to manage inside R).
I intend to make some short tutorials covering these many 'hidden' features of Rtrack.

Note also that the new polygonal object feature (currently only for NOR) should be used with care in combination with skewed objects as the 'de-skew' transformation also transforms the objects. Ideally the objects would be defined using the already-transformed arena. That should be easy to implement in an interactive web tool but will be complex to manage otherwise. Centred cameras are always a good idea to make life easier!  :)
